src/WK.OpenAiWrapper/Services/AssistantService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using OpenAI;
+using OpenAI.Assistants;
+
+namespace WK.OpenAiWrapper.Services;
+
+internal class AssistantService
+{
+    public async Task<bool> DeleteAssistantAsync(string assistantId)
+    {
+        using OpenAIClient client = new (Client.Instance.Options.Value.ApiKey);
+        return await client.AssistantsEndpoint.DeleteAssistantAsync(assistantId).ConfigureAwait(false);
+    }
+
+    public async Task<AssistantResponse> GetAssistantResponseByIdAsync(string assistantId)
+    {
+        using OpenAIClient client = new (Client.Instance.Options.Value.ApiKey);
+        return await client.AssistantsEndpoint.RetrieveAssistantAsync(assistantId).ConfigureAwait(false);
+    }
+    
+    public async Task<AssistantResponse> GetOrCreateAssistantResponse(string assistantName, CreateAssistantRequest assistantRequest)
+    {
+        try
+        {
+            using OpenAIClient client = new (Client.Instance.Options.Value.ApiKey);
+
+            ListResponse<AssistantResponse> assistantsResponse = await client.AssistantsEndpoint.ListAssistantsAsync().ConfigureAwait(false);
+            var assistantResponse = assistantsResponse.Items.SingleOrDefault(a => a.Name == assistantName)
+                                    ?? await client.AssistantsEndpoint.CreateAssistantAsync(assistantRequest).ConfigureAwait(false);
+            return assistantResponse;
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e);
```

Comment:
Was this just needed for debugging and is left over or is this really intended for production this way?

Reply:
That's what I'm asking myself right now. Must have had a reason at some point. I've removed it now.